### PR TITLE
Recurse ProcessTemplate call in case template file has further templates

### PR DESCRIPTION
**Pull request by rmjarvis**
@cwwalter pointed out that the template processing doesn't work correctly if the template file itself has a template field.  This PR fixes that.

### CODE REVIEW

**Comment by cwwalter** [comment]
Looks good.  Thanks!  So this can go into 2.4 too?

**Comment by rmjarvis** [comment]
Yes, I realized we can put this on the 2.4 series.  And can you turn your "looks good" into an actual review please?

File: tests/config_input/template.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  e2: 0
+
+input.power_spectrum.1.grid_spacing: 2
+
+gal.magnify:
+  type: PowerSpectrumMagnification
+  num: 1
+
+gal.ellip:
+  template: multirng:psf.ellip
```
Comment (cwwalter) [inline tests/config_input/template.yaml:41]: This is good, but just a note that it doesn't explicitly test the thing I noted:

```
file1.yaml
  template: mytemplate.yaml

file2.yaml
  template: file1.yaml
```

Although from reading the code I think it is handled the same.

File: tests/test_config_image.py [hunk 2]
```diff
@@ -2455,6 +2460,29 @@ def test_template():
...
+    config8 = galsim.config.ReadConfig('config_input/template.yaml')[0]
+    galsim.config.ProcessAllTemplates(config8)
+    for key in config3:
+        print(key)
+        print(config4[key])
+        print(config8[key])
+        assert config8[key] == config4[key]
+    assert config8 == config4
+
+    # Make sure nested templating works
```
Comment (cwwalter) [inline tests/test_config_image.py:2473]: I think the section above that tests the config_input/template.yaml file also tests nesting.. might want to note it above also.

File: tests/config_input/template.yaml (hunk 1 above)
Comment (cwwalter) [inline tests/config_input/template.yaml:41]: This is good, but just a note that it doesn't explicitly test the thing I noted that didn't work:

```
file1.yaml
  template: mytemplate.yaml

file2.yaml
  template: file1.yaml
```

Although from reading the code I think it is handled the same.

**Review by cwwalter — APPROVED** [review]
LGTM. 

I left just a few comments about things that were a bit unclear to me.

File: tests/test_config_image.py (hunk 2 above)
Comment (rmjarvis) [reply tests/test_config_image.py:2473]: No.  The previous test just confirms that template.yaml is equivalent to the thing we already tested above in python code.  config9 is the one that tests your use case of nesting templates.

File: tests/config_input/template.yaml (hunk 1 above)
Comment (rmjarvis) [reply tests/config_input/template.yaml:41]: That's tested by config9, not this template.yaml.  I thought it gratuitous to make another file that does this, but config9 does it in code.

Comment (cwwalter) [reply tests/config_input/template.yaml:41]: OK got it.  Thanks.  LGTM.

**Pull request merged**